eventuate-adapter-vertx/src/main/scala/com/rbmhtechnology/eventuate/adapter/vertx/VertxAdapterSystem.scala

```
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate.adapter.vertx
+
+import akka.actor.{ ActorSystem, Props }
+import com.rbmhtechnology.eventuate.adapter.vertx.api._
+import com.rbmhtechnology.eventuate.adapter.vertx.japi.rx.{ StorageProvider => RxStorageProvider }
+import com.rbmhtechnology.eventuate.adapter.vertx.japi.{ StorageProvider => JStorageProvider, VertxAdapterSystemConfig => JVertxAdapterSystemConfig }
+import io.vertx.core.Vertx
+import io.vertx.rxjava.core.{ Vertx => RxVertx }
+
+import scala.collection.immutable.Seq
+
+object VertxAdapterSystem {
+
+  import VertxConverters._
+
+  def apply(config: VertxAdapterSystemConfig, vertx: Vertx, storageProvider: StorageProvider)(implicit system: ActorSystem): VertxAdapterSystem =
+    new VertxAdapterSystem(config, vertx, storageProvider)
+
+  def create(config: JVertxAdapterSystemConfig,
+    vertx: Vertx,
+    storageProvider: JStorageProvider,
+    system: ActorSystem): VertxAdapterSystem =
+    new VertxAdapterSystem(config.underlying, vertx, storageProvider.asScala)(system)
+
+  def create(config: JVertxAdapterSystemConfig,
+    vertx: RxVertx,
+    storageProvider: RxStorageProvider,
+    system: ActorSystem): VertxAdapterSystem =
+    new VertxAdapterSystem(config.underlying, vertx, storageProvider.asScala)(system)
+}
+
+class VertxAdapterSystem private[vertx] (config: VertxAdapterSystemConfig, vertx: Vertx, storageProvider: StorageProvider)(implicit system: ActorSystem) {
+
+  private def registerEventBusCodecs(): Unit = {
+    config.codecClasses.foreach(c =>
+      try {
+        vertx.eventBus().registerDefaultCodec(c.asInstanceOf[Class[AnyRef]], AkkaSerializationMessageCodec(c))
+      } catch {
+        case e: IllegalStateException =>
+          system.log.warning(s"An adapter codec for class ${c.getName} was configured, even though a default codec was already registered for this class.")
```

Shouldn't we propagate this exception to the application?